latent_programmer/decomposition_transformer_attention/decomposition_models.py

```diff
@@ -223,4 +258,22 @@ def __call__(self,
     encoded = self.encode(inputs, outputs)
     encoded_padding_mask = jnp.where(outputs > 0, 1, 0).astype(jnp.float32)
 
+    if self.has_variable('relative_attention', 'max_input_distance'):
+      max_input_distance = jnp.max(jnp.count_nonzero(inputs == 0, axis=0))
```

Comment:
is `jnp.count_nonzero(array == 0, axis=0)` correct? i think instead we want `jnp.count_nonzero(array != 0, axis=1) - 1`, where the -1 is because we want distance not the number of tokens. but for programs, we need -2 since the last element is EOS which won't produce a distance